### PR TITLE
Feature/youtube apis gsd 1890

### PR DESCRIPTION
**Pull request by aashishAil**
## Relevant Documentation Links
- https://creatorstack.atlassian.net/wiki/spaces/PF/pages/629309585/YouTube+Video+Scraping+v2.0

## Relevant Jira Links
- https://creatorstack.atlassian.net/browse/GSD-1890

## Other Relevant PRs
- 

## Changes
- Added common youtube data fetch API's
- Added generic logger 
- Added generic new-relic profiler

### Changes Include

-   [ ] Bug fix 🐛 (non-breaking change which fixes an issue)
-   [ ] New feature ✨ (non-breaking change which adds functionality)
-   [ ] Breaking change 😢 (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)
-   [ ] Tooling ⚙️ (Changes to the build process or auxiliary tools and libraries such as documentation generation)
-   [ ] Performance ⚡ (code change that improves performance)

### Checklist

-   [ ] I have commented on my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] Requires migrations
-   [ ] Requires dependency update
-   [ ] Breaking API Change
-   [ ] I have added proper logs for production debugging
-   [ ] I have added performance (APM) monitoring checkpoints
-   [ ] I have added unit tests

## Screenshots (if any)
(prefer animated gif)

### Test #1

Description

- [ ] Acceptance Creteria #1
- [ ] Acceptance Creteria #2

### CODE REVIEW

**Review by mrasif — APPROVED** [review]

File: repository/youtube.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	"net/http"
+	"strconv"
+
+	"github.com/CreatorStack/backbone/constants"
+	"github.com/CreatorStack/backbone/logger"
+	"github.com/CreatorStack/backbone/models"
+	"github.com/CreatorStack/backbone/utils"
+)
+
+type YoutubeRepo interface {
```
Comment (MrDHat) [inline repository/youtube.go:19]: shouldn't this be in the youtube service?

**Review by MrDHat — CHANGES_REQUESTED** [review]

File: repository/youtube.go (hunk 1 above)
Comment (aashishAil) [reply repository/youtube.go:19]: This is common for both Fanclub and youtube service, hence moving it here

Comment (aashishAil) [reply repository/youtube.go]: This has been removed

File: utils/profiler.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+	trackerType         int
+	interaction         *newrelic.Segment
+	databaseInteraction *newrelic.DatastoreSegment
+	networkInteraction  *newrelic.ExternalSegment
+	queueInteraction    *newrelic.MessageProducerSegment
+	transaction         *newrelic.Transaction
+}
+
+const (
+	INTERACTION = iota
```
Comment (krishnamurthypranesh) [inline utils/profiler.go:42]: Question: Is `iota` coming from an import?

**Review by krishnamurthypranesh — APPROVED** [review]

File: utils/profiler.go (hunk 2 above)
Comment (aashishAil) [reply utils/profiler.go:42]: this is for new-relic specifically. Their convention

**Pull request merged**